packages/testing/playwright/tests/ui/subworkflow-wait.spec.ts

```diff
@@ -0,0 +1,74 @@
+			await api.workflows.importWorkflowFromFile('subworkflow-wait-child.json')
+		).workflowId;
+
+		const filePath = resolveFromRoot('workflows', 'subworkflow-parent-no-wait.json');
+		const fileContent = readFileSync(filePath, 'utf8');
+		const workflowDefinition = JSON.parse(fileContent) as IWorkflowBase;
+		expect(workflowDefinition?.nodes[0]?.parameters).toBeDefined();
+		// Replace the placeholder workflow ID with the actual child workflow ID
+		workflowDefinition.nodes[0].parameters.workflowId = {
+			value: childWorkflowId,
+			mode: 'list',
+		};
+		const { webhookPath, workflowId } =
+			await api.workflows.importWorkflowFromDefinition(workflowDefinition);
+
+		const response = await api.request.get(`/webhook/${webhookPath}`);
+		expect(response.ok()).toBe(true);
+		const execution = await api.workflows.waitForExecution(workflowId, 5000);
+		expect(execution.status).toBe('success');
+
+		// The child workflow should still be running or waiting, since it's configured to wait 120s.
+		const getExecutionsResponse = await api.workflows.getExecutions(childWorkflowId);
+		// TODO: figure out why the filtering in `getExecutions` isn't working.
+		const childExecutions = getExecutionsResponse.filter((e) => e.workflowId === childWorkflowId);
+		expect(childExecutions.length).toBe(1);
+		expect(childExecutions[0].status).toMatch(/running|waiting/);
+	});
+
+	test('CAT-1445 should not be restarted by the child workflow finishing', async ({ api }) => {
+		// The child is a no-op that returns immediately.
+		const childWorkflowId = (
+			await api.workflows.importWorkflowFromFile('subworkflow-noop-child.json')
+		).workflowId;
+
+		// This is a parent that does NOT wait for the child to finish, but it has its own separate Wait node.
+		// We want to verify that the parent is NOT restarted when the child finishes. This was fixed in CAT-1445.
+		const filePath = resolveFromRoot('workflows', 'subworkflow-waiting-parent-no-child-wait.json');
+		const fileContent = readFileSync(filePath, 'utf8');
+		const workflowDefinition = JSON.parse(fileContent) as IWorkflowBase;
+		expect(workflowDefinition?.nodes[1]?.parameters).toBeDefined();
+		// Replace the placeholder workflow ID with the actual child workflow ID
+		workflowDefinition.nodes[1].parameters.workflowId = {
+			value: childWorkflowId,
+			mode: 'list',
+		};
+		const { webhookPath, workflowId } =
+			await api.workflows.importWorkflowFromDefinition(workflowDefinition);
+
+		const response = await api.request.get(`/webhook/${webhookPath}`);
+		expect(response.ok()).toBe(true);
+
+		// First, wait for the child to finish.
+		const childExecution = await api.workflows.waitForExecution(childWorkflowId, 5000);
+		expect(childExecution.status).toBe('success');
+
+		// Now check if the parent is finished. It should be waiting.
+		// We need to wait a bit for the parent to transition to waiting state after the child completes
+		await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
We have `retryUntil` in `packages/cli/test/integration/shared/retry-until.ts`. We could have a similar helper in the playwright package. Then we wouldn't need the explicit wait and this would continue as soon as the assertion succeeds.

I think I mentioned this same to Declan in some other PR but I can't find any helper in playwright package, so maybe he didn't add it 🤔